airflow/providers/amazon/aws/links/emr.py

```diff
@@ -33,3 +40,23 @@ class EmrLogsLink(BaseAwsLink):
     name = "EMR Cluster Logs"
     key = "emr_logs"
     format_str = BASE_AWS_CONSOLE_LINK + "/s3/buckets/{log_uri}?region={region_name}&prefix={job_flow_id}/"
+
+
+def get_log_uri(
+    *, cluster: dict[str, Any] | None = None, emr_client: boto3.client = None, job_flow_id: str | None = None
+) -> str:
+    """
+    Retrieves the S3 URI to the EMR Job logs.  Requires either the output of a
+    describe_cluster call or both an EMR Client and a job_flow_id to look it up.
+    """
+    if not exactly_one(cluster, emr_client and job_flow_id):
```

Comment:
Could also do an XOR `bool(cluster) ^ bool(emr_client and job_flow_id)` here, but the `exactly_one` helper was handy and more clear IMHO.

Reply:
lgtm

Reply:
Yeah, thanks.   The more I look at it the more I think the "clarifying" parens are even more confusing.  I'll leave it like this.